### PR TITLE
chore: controller deployment rename and more docs.

### PR DESCRIPTION
**Pull request by whynowy**
1. Controller deployment renamed to `numaflow-controller`;
2. Doc names changed to lower cases - it's too hard to always use upper cases.
3. More docs.

### CODE REVIEW

File: docs/debugging.md [hunk 1]
```diff
@@ -22,14 +22,14 @@ spec:
         env:
           - name: NUMAFLOW_DEBUG
             value: "true" # DO NOT forget the double quotes!!!
-    - name: output
+    - name: out
       sink:
         log: {}
   edges:
-    - from: input
+    - from: in
```
Comment (vigith) [inline docs/debugging.md:29]: shouldn't this be `inp`?

File: docs/environment-variables.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Environment Variables
+
+For the `main` container of vertex pods, environment variable `NUMAFLOW_DEBUG` can be set to `"true"` for [debugging](./debugging.md).
```
Comment (vigith) [inline docs/environment-variables.md:3]: should we callout explicitly that the `true` should be in `"` ?

**Review by vigith — APPROVED** [review]

File: docs/debugging.md (hunk 1 above)
Comment (whynowy) [reply docs/debugging.md:29]: oops.. updated.

File: docs/environment-variables.md (hunk 2 above)
Comment (whynowy) [reply docs/environment-variables.md]: Removed the ", since it's been called out in the next linked page.

**Pull request merged**